server/server.go

```diff
@@ -168,3 +174,22 @@ func setupHttpClient(idpCaFile string) (*http.Client, error) {
 
 	return client, nil
 }
+
+func authVerify(middleware *samlsp.Middleware) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		session, err := middleware.Session.GetSession(r)
+
+		if session != nil {
+			w.WriteHeader(204)
+			return
+		}
+
+		if err == samlsp.ErrNoSession {
+			w.WriteHeader(401)
+			return
+		}
+
```

Comment:
Perhaps log any other error that makes it here?

Reply:
The function is unwrapped from crewjam/saml, I added the existing middleware error callback now.

Reply:
Now I see that is the only error they document returning there. So,  no further change needed?

Reply:
I don't believe so, tried to keep the same behaviour as the middleware.